cubedash/_stac.py

```diff
@@ -266,7 +266,40 @@ def as_stac_item(dataset: DatasetItem) -> pystac.Item:
     return item
 
 
+def _provider_from_dict(d: dict) -> pystac.Provider:
+    """A custom parser that extends pystac.Provider.from_dict()"""
+    # pystac.Provider.from_dict() does not validate roles against ProviderRole nor create a list
```

Comment:
What does "nor create a list" mean? The type signature for `Provider.role` is `list[ProviderRole] | None` according to the pystac documentation.

Reply:
Updated comment to help clarify. My intention was that this is a reasonable place to ensure we're passing on values/types that pystac requires. We need to validate role values anyway (because `pystac.Provider` doesn't) so thought we could add a couple of convenience mappings too .. e.g., we can pass in a single string in `provider["roles"]=string`, rather than knowing/enforcing earlier that this element should be `provider["roles"]=[string]`.